### ISSUE TITLE
Incorrect "Next Check-In" when target system clock is not up to date

### ISSUE THREAD

**Issue by mrThe** [BishopFox/sliver#684]
**Describe the bug**
Minor issue when target clock is not up to date. In beacons list in column "Next Check-In" it will show weird things like "-47m35.154999998s"

**To Reproduce**
1. Run sliver
2. Generate beacon
3. Run beacon on host with not synced time

**Expected behavior**
Don't rely on target datetime, and calculate next checking using server time, so it will be correct.

**Screenshots**
<img width="432" alt="image" src="https://user-images.githubusercontent.com/1254163/169702169-dc6a9ee8-2cb0-4310-9658-9ae36cf4cb0f.png">

**Sliver:**
[*] Client v1.5.12 - aacf2c16ac00e4609231f5faee588bdb9ea9c532 - linux/amd64
    Compiled at 2022-04-20 20:06:30 +0000 UTC
    Compiled with go version go1.18 linux/amd64

[*] Server v1.5.12 - aacf2c16ac00e4609231f5faee588bdb9ea9c532 - linux/amd64
    Compiled at 2022-04-20 20:06:30 +0000 UTC

**Comment by mrThe** [comment]
Well, i guess after #685 there is another issue for that case..
<img width="658" alt="image" src="https://user-images.githubusercontent.com/1254163/171023173-a78912ed-2ddb-4cf0-a64e-98efa42ce4f0.png">
:) 

Beacon works as expected tho

**Comment by moloch--** [comment]
Yea, i think @RafBishopFox pushed this change.

**Comment by RafBishopFox** [comment]
I am not sure how to preserve backwards compatibility in this case. There is not enough information in the beacon registration message to differentiate an "old" beacon from a "new" one.

Previously, the beacon registration message included a Unix timestamp that represented it was scheduled to check in next. The calculation happened on the implant side and that is what resulted in the issue you saw. My PR changed things so that the implant reported how many seconds until its next check in. This made implants generated since the change unaffected by an implant with an out of sync clock on the implant side.

For backwards compatibility, we could look at the next check in number provided by the implant. If we treat it as a timestamp and the time occurs in the future, then the implant is probably "old." The behavior will be the same as before though, so it will not fix the issue for "old" implants, but at least new implants will be taken care of. I guess that will break things with a new implant if the beacon interval is greater than 1,653,969,600 seconds (that is the timestamp for May 31, 2022 at 00:00:00 UTC). I doubt there are too many implants out there with that interval, so we are probably safe. That should be a relatively easy change to make.

@mrThe - What was the beacon you generated to see the behavior you saw in your update? I did not see any of my test implants with the same last check in time and next check in time.

**Comment by moloch--** [comment]
I think the slight backwards incompatibility is fine so long as it's not crashing.

**Comment by RafBishopFox** [comment]
Yeah - this is only affecting the display of the beacon time, not the actual beacon itself.

**Comment by mrThe** [comment]
@RafBishopFox  It was just a regular beacon for x64 linux, with all default settings, except for mtls connection address ofc. Generated on version mentioned in first message. 

Well, technically i don't think that back compatibility is such a big deal, since it doesn't break anything critical.
Also, the beacon from the second screenshot are still running, and it works as expected, showing the right time, so i guess it's safe to close this issue.
